### PR TITLE
[16.0][FIX] sale_order_type: Avoid unwanted triggering of compute functions

### PR DESCRIPTION
**Pull request by grindtildeath**
Having account.move._compute_journal_id overriden with an added dependency to sale_type_id is messing with computation triggers, because compute functions depending on journal_id will get recomputed, potentially overwriting existing value, as is the case with bank account on account.payment.

Use instead the hook to search for default journal, in order to return the journal defined on sale order type to its compute function.

### CODE REVIEW

File: sale_order_type/models/account_move.py [hunk 1]
```diff
@@ -48,6 +48,11 @@ def _compute_sale_type_id(self):
                 )
                 if sale_type:
                     record.sale_type_id = sale_type
+            # Add manually field to compute, as using api.depends on its compute function would
+            #  trigger unwanted recomputation of partner_bank_id on vendor bills
+            self.env.add_to_compute(
+                self.env["account.move"]._fields["journal_id"], record
+            )
```
Comment (ivantodorovich) [inline sale_order_type/models/account_move.py:55]: Careful, AFAIK, when using `add_to_compute`, the field dependencies aren't recomputed unless you explicitly call `modified`, too.

**Review by henrybackman — APPROVED** [review]

**Comment by grindtildeath** [comment]
Since Odoo accepted my fix and fast tracked it, I'm not sure if this change is still needed :question:

File: sale_order_type/models/account_move.py (hunk 1 above)
Comment (rousseldenis) [reply sale_order_type/models/account_move.py:55]: IMHO, this is more a bad core architecture as `partner_bank_id` field is computed/stored and not its dependency `bank_partner_id`.

Comment (rousseldenis) [reply sale_order_type/models/account_move.py:55]: Have you tried to change the depends of `partner_bank_id`field using `commercial_partner_id`instead ?

https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_move.py#L1315

**Comment by grindtildeath** [comment]
IMO we can close since https://github.com/odoo/odoo/pull/181278 is merged

**Pull request closed without merging**